src/Uns.ts

```diff
@@ -43,33 +74,23 @@ export default class Uns extends NamingService {
 
   constructor(source?: UnsSource) {
     super();
-    if (
-      source &&
-      source.locations &&
-      (!source.locations.Layer1 || !source.locations.Layer2)
-    ) {
-      throw new ConfigurationError(
-        ConfigurationErrorCode.NetworkConfigMissing,
-        {
-          method: NamingServiceName.UNS,
-          config: !source.locations.Layer1 ? 'Layer1' : 'Layer2',
-        },
-      );
-    }
-    if (!source) {
+    if (source) {
+      ensureValidSourceConfig(source);
+    } else {
```

Comment:
Is the `else` statement still necessary? Looks like the `source` param will always available? We can either update the function param to require `source` or just throw an error.

Reply:
yea this would be ideal, but we still have the concept of UdApi. 
So when someone config with UdApi, it supposed to configure a fake Uns object support methods like `namehash` https://github.com/unstoppabledomains/resolution/blob/master/src/Resolution.ts#L77-L89. That's why I still have to allow empty source here

Reply:
Ah ok. Sounds good